crates/bevy_ui/src/lib.rs

```diff
@@ -289,26 +290,27 @@ fn build_text_interop(app: &mut App) {
         PostUpdate,
         (
             (
-                bevy_text::detect_text_needs_rerender::<Text>,
+                bevy_text::detect_text_root_needs_rerender::<Text>,
                 widget::measure_text_system,
             )
                 .chain()
                 .in_set(UiSystems::Content)
                 // Text and Text2d are independent.
-                .ambiguous_with(bevy_text::detect_text_needs_rerender::<bevy_text::Text2d>)
+                .ambiguous_with(bevy_text::detect_text_root_needs_rerender::<bevy_text::Text2d>)
                 // Potential conflict: `Assets<Image>`
                 // Since both systems will only ever insert new [`Image`] assets,
                 // they will never observe each other's effects.
                 .ambiguous_with(bevy_text::update_text2d_layout)
                 // We assume Text is on disjoint UI entities to ImageNode and UiTextureAtlasImage
                 // FIXME: Add an archetype invariant for this https://github.com/bevyengine/bevy/issues/1481.
-                .ambiguous_with(widget::update_image_content_size_system),
+                .ambiguous_with(widget::update_image_content_size_system)
+                .after(detect_text_span_needs_rerender),
```

Comment:
This means text span rerender detection is not locked into the `UiSystems` system sets. Which means systems in `PostUpdate` that deal with text stuff `.before(UiSystems::Prepare)` could be ordered wrong relative to text span rerender detection.

Can you move `detect_text_span_needs_rerender` to a system set then in this plugin do `.configure_sets(PostUpdate, TextSpanDetectSet.in_set(UiSystems::Content)`?

Reply:
> Although that may screw up `text2d` ordering. Which is might be why I originally allowed duplicate text span change detection.

I guess we could work around this a different way, maybe by adding a `Has<Root>` to the `computed` query? Walking all the text spans twice though feels really unpleasant though, even if it's not that expensive in practice.